tests/unit/services/hifi-test.js

```diff
@@ -811,3 +826,179 @@ test("metadata can be sent with a play and load request and it will stay with th
+      done();
+    });
+  });
+});
+
+test("new-load-request gets fired on new load requests that are cached", function(assert) {
+  let done        = assert.async();
+  let connections = ['NativeAudio'];
+  let service     = this.subject({ options: chooseActiveConnections(...connections) });
+  let s1url       = "/assets/silence.mp3";
+
+  assert.expect(4);
+
+  service.one('new-load-request', ({urlsOrPromise, options}) => {
+    assert.equal(urlsOrPromise, s1url, "url should equal url passed in");
+    assert.equal(options.metadata.id, 1, "metadata id should be equale");
+  });
+
+  return service.load(s1url, {metadata: {id: 1}}).then(() => {
+    service.one('new-load-request', ({urlsOrPromise, options}) => {
+      assert.equal(urlsOrPromise, s1url, "url should equal url passed in");
+      assert.equal(options.metadata.id, 2, "metadata id should be undefined");
+    });
+
+    return service.load(s1url, {metadata: {id: 2}}).then(() => {
+      done();
+    });
+  });
+});
+
+test("audio-position-will-change gets fired on position changes", function(assert) {
+  let done        = assert.async();
+  let service     = this.subject({ options: activateDummyConnection() });
+  let s1url       = "/good/15000/test";
+
+  assert.expect(2);
+
+  service.one('audio-position-will-change', (sound, {currentPosition, newPosition}) => {
+    assert.equal(currentPosition, 0, "current position should be zero");
+    assert.equal(newPosition, 5000, "new position should be 5000");
+  });
+
+  return service.play(s1url).then(() => {
+    service.set('position', 5000);
+    done();
+  });
+});
+
+test("audio-will-rewind gets fired on rewind", function(assert) {
+  let done        = assert.async();
+  let service     = this.subject({ options: activateDummyConnection() });
+  let s1url       = "/good/15000/test2";
+
+  assert.expect(4);
+
+  service.one('audio-will-rewind', (sound, {currentPosition, newPosition}) => {
+    assert.equal(currentPosition, 5000, "current position should be zero");
+    assert.equal(newPosition, 4000, "new position should be 5000");
+  });
+
+  return service.play(s1url, {position: 5000}).then(() => {
+    service.rewind(1000);
+
+    service.on('audio-will-rewind', (sound, {currentPosition, newPosition}) => {
+      assert.equal(currentPosition, 4000, "current position should be zero");
+      assert.equal(newPosition, 0, "new position should be 5000");
+    });
+
+    service.rewind(6000);
+    done();
+  });
+});
+
+test("audio-will-fast-forward gets fired on fast forward", function(assert) {
+  let done        = assert.async();
+  let service     = this.subject({ options: activateDummyConnection() });
+  let s1url       = "/good/15000/1.mp3";
+
+  assert.expect(2);
+
+  service.on('audio-will-fast-forward', (sound, {currentPosition, newPosition}) => {
+    assert.equal(currentPosition, 5000, "current position should be zero");
+    assert.equal(newPosition, 6000, "new position should be 5000");
```

Comment:
these messages are wrong, I believe. `"current position should be *5000*"` and `"new position should be *6000*"`